antenna/app.py

```diff
@@ -5,128 +5,34 @@
-from everett.component import ConfigOptions, RequiredConfigMixin
+from everett.manager import (
+    ConfigManager,
+    ConfigOSEnv,
+    get_config_for_class,
+    Option,
+)
 import falcon
-import markus
 
 from antenna.breakpad_resource import BreakpadSubmitterResource
+from antenna.crashmover import CrashMover
 from antenna.health_resource import (
     BrokenResource,
     HeartbeatResource,
     LBHeartbeatResource,
     VersionResource,
 )
 from antenna.heartbeat import HeartbeatManager
+from antenna.liblogging import setup_logging, log_config
+from antenna.libmarkus import setup_metrics
 from antenna.sentry import (
     set_sentry_client,
     setup_sentry_logging,
     wsgi_capture_exceptions,
 )
 
 
-logger = logging.getLogger(__name__)
-
-
-_LOGGING_SETUP = False
-
-
-def setup_logging(app_config):
-    """Initialize Python logging configuration."""
-    global _LOGGING_SETUP
-    if _LOGGING_SETUP:
-        # NOTE(willkg): This makes it so that logging is only set up once per process.
-        return
-
-    host_id = app_config("host_id") or socket.gethostname()
-
-    class AddHostID(logging.Filter):
-        def filter(self, record):
-            record.host_id = host_id
-            return True
-
-    dc = {
-        "version": 1,
-        "disable_existing_loggers": True,
-        "filters": {"add_hostid": {"()": AddHostID}},
-        "formatters": {
-            "socorroapp": {
-                "format": "%(asctime)s %(levelname)s - %(name)s - %(message)s"
-            },
-            "mozlog": {
-                "()": "dockerflow.logging.JsonLogFormatter",
-                "logger_name": "antenna",
-            },
-        },
-        "handlers": {
-            "console": {
-                "level": "DEBUG",
-                "class": "logging.StreamHandler",
-                "formatter": "socorroapp",
-                "filters": ["add_hostid"],
-            },
-            "mozlog": {
-                "level": "DEBUG",
-                "class": "logging.StreamHandler",
-                "formatter": "mozlog",
-                "filters": ["add_hostid"],
-            },
-        },
-        "loggers": {
-            "antenna": {
-                "propagate": False,
-                "handlers": ["mozlog"],
-                "level": app_config("logging_level"),
-            }
-        },
-        "root": {"handlers": ["mozlog"], "level": "WARNING"},
-    }
-
-    if app_config("local_dev_env"):
-        # In a local development environment, we log to the console in a human-readable
-        # fashion and add a markus logger
-        dc["loggers"]["antenna"]["handlers"] = ["console"]
-        dc["loggers"]["markus"] = {
-            "propagate": False,
-            "handlers": ["console"],
-            "level": "INFO",
-        }
-        dc["root"]["handlers"] = ["console"]
-
-    logging.config.dictConfig(dc)
-    _LOGGING_SETUP = True
-
-
-def setup_metrics(metrics_classes, config, logger=None):
```

Comment:
I redid how this work so we're not doing metrics classes with their own setup but rather just setting up markus directly. That code is now in `libmarkus.py`.